src/redis-server-helper.js

```diff
@@ -0,0 +1,95 @@
+import wait from './wait';
+import runCommand from './run-command';
+
+function containsValidPort(args) {
+	const flag = args.indexOf('--port');
+	if (flag > -1 && flag + 1 < args.length) {
+		const port = args[flag + 1];
+		if (typeof port !== 'number' || port < 0 || port > 65535) {
+			return false;
+		}
+	}
+	return true;
+}
+
+function getPort(args) {
+	const flag = args.indexOf('--port');
+	return flag > -1 ? args[flag + 1] : 6379;
+}
+
+async function checkIfReady(args) {
+	const port = getPort(args);
+
+	try {
+		const { childProcess } = await runCommand('redis-cli', [
+			'-p', port,
+			'-r', '1', 'PING',
+		]);
+
+		const { stdout } = await childProcess;
+		return stdout === 'PONG';
+	} catch (err) {
+		return false;
+	}
+}
+
+async function waitUntilReady(tries, interval, backoff, args) {
+	if (tries <= 0) throw new Error('Timed out');
+
+	await wait(interval);
+
+	const ready = await checkIfReady(args);
+	if (ready) return true;
+
+	return waitUntilReady(tries - 1, interval * backoff, backoff, args);
+}
+
+export class RedisServerHelper {
+	constructor(args, opts) {
```

Comment:
```suggestion
	constructor(args = [], { downloadDir, version } = {}) {
```

This lets us delete most of the `args` and `opts` validation.